WIP: Catch Directory.CreateDirectory() exceptions in New-Item Cmdlet for FileSystemProvider

xtqqczze opened the pull request:
Changes to FileSystemProvider.cs for the New-Item Cmdlet

## PR Summary

* Catch IOException and UnauthorizedAccessException when calling CreateDirectory()
* Check if a file exists at the path before calling CreateDirectory() (fixes #5290)
* Fix SA1005:SingleLineCommentsMustBeginWithSingleSpace

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

xtqqczze committed:
Update FileSystemProvider.cs

Catch Directory.CreateDirectory() exceptions

xtqqczze committed:
Fixed SA1005:SingleLineCommentsMustBeginWithSingleSpace

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/PowerShell/PowerShell?pullRequest=8532) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/PowerShell/PowerShell?pullRequest=8532) before we can accept your contribution.<br/><br/>:x: xtqqczze [sign now](https://cla.opensource.microsoft.com/PowerShell/PowerShell?pullRequest=8532)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/PowerShell/PowerShell?pullRequest=8532) it.</sub>

xtqqczze committed:
Update NewItem() for ItemType.Directory

Check if the path exists and is not a container before calling CreateDirectory()

xtqqczze committed:
Update ErrorRecord message

xtqqczze committed:
Update NewItem() for ItemType.File

Check the path is not a directory before creating a file

xtqqczze committed:
Rename variables
